src/material-experimental/mdc-slider/slider.ts

```diff
@@ -22,13 +75,229 @@ import {SliderAdapter} from './slider-adapter';
   styleUrls: ['slider.css'],
+  host: {
+    'class': 'mat-mdc-slider mdc-slider',
+    '[class.mdc-slider--range]': 'isRange',
+    '[class.mdc-slider--disabled]': 'isDisabled',
+    '[class.mdc-slider--discrete]': 'isDiscrete',
+    '[class.mdc-slider--tick-marks]': 'hasTickMarks',
+  },
+  exportAs: 'matSlider',
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
 })
-export class MatSlider {
+export class MatSlider implements AfterViewInit, OnDestroy {
+  /** The slider thumb(s). */
+  @ViewChildren('thumb') _thumbs: QueryList<ElementRef<HTMLElement>>;
 
-  /** Instance of the MDC slider foundation for this slider. */
+  /** The slider thumb knob(s) */
+  @ViewChildren('knob') _knobs: QueryList<ElementRef<HTMLElement>>;
+
+  /** The active section of the slider track. */
+  @ViewChild('trackActive') _trackActive: ElementRef<HTMLElement>;
+
+  /** The sliders hidden range input(s). */
+  @ContentChildren(MatSliderThumb, { descendants: false }) _inputs: QueryList<MatSliderThumb>;
+
+  /** Whether the slider is disabled. */
+  @Input()
+  get isDisabled(): boolean { return this._isDisabled; }
+  set isDisabled(v: boolean) {
+    this._isDisabled = coerceBooleanProperty(v);
+    if (this._initialized) {
+      this._foundation.setDisabled(v);
+    }
+  }
+  private _isDisabled: boolean = false;
+
+  /** Whether the slider displays a numeric value label upon pressing the thumb. */
+  @Input()
+  get isDiscrete(): boolean { return this._isDiscrete; }
+  set isDiscrete(v: boolean) { this._isDiscrete = coerceBooleanProperty(v); }
+  private _isDiscrete: boolean = false;
+
+  /** Whether the slider displays tick marks along the slider track. */
+  @Input()
+  get hasTickMarks(): boolean { return this._hasTickMarks; }
+  set hasTickMarks(v: boolean) { this._hasTickMarks = coerceBooleanProperty(this._hasTickMarks); }
+  private _hasTickMarks: boolean = false;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number { return this._min; }
+  set min(v: number) { this._min = coerceNumberProperty(v, this._min); }
+  private _min = 0;
+
+  /** The maximum value that the slider can have. */
+  @Input()
+  get max(): number { return this._max; }
+  set max(v: number) { this._max = coerceNumberProperty(v, this._max); }
+  private _max = 100;
+
+  /** The values at which the thumb will snap. */
+  @Input()
+  get step(): number { return this._step; }
+  set step(v: number) { this._step = coerceNumberProperty(v, this._step); }
+  private _step: number = 1;
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: ((value: number) => string) | null;
+
+  /** Event emitted when a slider thumb value has changed. */
+  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
```

Comment:
I was just matching the old API. I can remove them if you'd like

Reply:
It's been a while since we discussed the API. I _think_ we decided that we were going to try to keep most of the API but split it between the container and thumbs. That way people would need to make some changes, but hopefully they would be fairly straightforward, just adding the wrapper and moving a few properties to it. @jelbourn is that what you remember too?

Reply:
Let's go with that since there are relatively few usages in Google so it should be pretty straightforward to swap